azure-quantum/azure/quantum/job/session.py

```diff
@@ -73,54 +99,124 @@ def __init__(
 
     @property
     def details(self) -> SessionDetails:
+        """Get the session details.
+        :return: The details about the session.
+        :rtype: SessionDetails
+        """
         return self._details
 
     @details.setter
     def details(self, value: SessionDetails):
+        """Set session details.
+        :param value: The details about the session
+        :type value: SessionDetails
+        """
         self._details = value
 
     @property
     def target(self) -> "Target":
+        """Get the target associated with the session.
+        :return: The target associated with the session.
+        :rtype: Target
+        """
         return self._target
 
     def open(self) -> "Session":
+        """Opens a session, effectively creating a new session in the
+           Azure Quantum service, and allowing it to accept jobs under it.
+
+        :return: The session object with updated details after its opening.
+        :rtype: Session
+        """
         self.workspace.open_session(self)
         return self
 
     def close(self) -> "Session":
+        """Closes a session, not allowing further jobs to be submitted under
+           the session.
+
+        :return: The session object with updated details after its closing.
```

Comment:
Does the autogenerated client take care of updating itself based on response?

Reply:
the autogenerated client does not, but the `workspace.close_session()` does
```
session.details = client.close(session_id=session.id)
```